### DISCUSSION TITLE
With what flags should I compile a dll to be able safely resize it?

### DISCUSSION THREAD

**Discussion by wer14** [avast/pe_tools#8] [Q&A]
When trying to edit a dll the exception raises up:

```
if not pe.is_dir_safely_resizable(IMAGE_DIRECTORY_ENTRY_RESOURCE):
            raise(Exception('error: the resource section is not resizable: {}'.format(file))) 
```

Maybe there is some flags to make dll safely resizable?

**Comment by avakar** [comment]
This is a sanity check to make sure that in the event the resource section needs to be made bigger, all sections following the resources will survive being moved to a higher address.

If you compile with MSVC, the resource section will only be followed by relocations, which can be moved arbitrarily. The tool will recognize the reloc section and won't trigger the exception. No extra flags are needed.

I'm guessing you're compiling with clang? Can you share the binary? If not, what sections follow the resource section in your case?

You can either get rid of those immovable sections, or, if they're movable, we can update the tool to support them.

**Reply by wer14** [reply]
Thank you for your quick reply. We are building with MSVC. The problem binary was qmake.exe. Apparently it does not have .rsrc section.
The problem was solved by excluding the binary from editing.

**Reply by avakar** [reply]
Ah, the message is rather confusing in this case. I've opened #9 to address it. Thanks!

**Comment by mikkeljohnsen** [comment]
@gmikhaile @avakar 
"The problem was solved by excluding the binary from editing." what do you mean by that ?

I'm trying to set on a exe file, created with "dotnet publish". Is that possible ?

**Reply by wer14** [reply]
> what do you mean by that ?

We are skipping qmake.exe so the binary will not be edited.
